### PR TITLE
Membership tiers

### PR DESCRIPTION
**Pull request by millerf**
# OLD - Rebased in https://github.com/Automattic/wp-calypso/pull/82728

Fixes to https://github.com/Automattic/gold/issues/140
Related to https://github.com/Automattic/jetpack/pull/32658
And D119740-code

## Proposed Changes

* Change plan list according to Figma
* Add Newsletter tier to bundle 2 plans
* Change the add/edit modal to handle such changes
* Lock the Newsletter selector when a plan was added as-is.
* It should be retro-compatible with existing plans.
* Display only one success notice for tiers
* Fix: only show "you can now add your first payment plan" if needed  (cannot be tested as Stripe sends you back to WP.com)

| list | modal |
| ---- | ---- |
|  Screenshot 2023-08-24 at 19 44 08 |  Screenshot 2023-08-24 at 19 44 17 | 

## Testing Instructions

* Patch D119740-code on your sandbox
* Add new plans, delete and modify to test
* Have a site with a monthly newsletter plan and a yearly newsletter plan. Try to edit them with this patch on. It should created bundled plans.

CC @crisbusquets for designs recitifications

### CODE REVIEW

**Comment by pottedmeat** [comment]
Testing on Calypso for a Jetpack Ninja site:

- I created Gold and Silver plans.
- I updated the Gold plan by tripling the priced to 3/36
- It created a new monthly plan with the yearly price.

image

**Comment by millerf** [comment]
> Testing on Calypso for a Jetpack Ninja site:
> 
> * I created Gold and Silver plans.
> * I updated the Gold plan by tripling the priced to 3/36
> * It created a new monthly plan with the yearly price.
> 
> image

I am not sure what I am seeing here.

> I created Gold and Silver plans.

Those were creating before the patch? It looks like they have a monthly plan but no yearly plans attached, which should not be possible when creating plans/tiers with this PR...

**Comment by pottedmeat** [comment]
I was able to test succesfully.

**Comment by millerf** [comment]
> I was able to test succesfully.

Did you only test this PR or the https://github.com/Automattic/jetpack/pull/32710?

**Comment by pottedmeat** [comment]
> Did you only test this PR or the https://github.com/Automattic/jetpack/pull/32710?

I followed the Jetpack PR instructions.

File: client/my-sites/earn/components/add-edit-plan-modal/index.tsx [hunk 1]
```diff
@@ -87,8 +90,11 @@ const MAX_LENGTH_CUSTOM_CONFIRMATION_EMAIL_MESSAGE = 2000;
 const RecurringPaymentsPlanAddEditModal = ( {
 	closeDialog,
 	product,
+	annualProduct /* annual product for tiers */,
 	siteId,
 }: RecurringPaymentsPlanAddEditModalProps ) => {
+	const monthlyFrequency = '1 month';
+	const annualFrequency = '1 year';
```
Comment (n3f) [inline client/my-sites/earn/components/add-edit-plan-modal/index.tsx]: Should these be promoted to consts outside the component?

e.g.

```js
const MONTHLY_FREQUENCY = '1 month';
const YEARLY_FREQUENCY = '1 year';
...
const RecurringPaymentsPlanAddEditModal = ( {
```

**Review by n3f — APPROVED** [review]
WFM 👍 (plus a small comment)

File: client/my-sites/earn/components/add-edit-plan-modal/index.tsx (hunk 1 above)
Comment (millerf) [reply client/my-sites/earn/components/add-edit-plan-modal/index.tsx]: Yep. Nice one

Comment (millerf) [reply client/my-sites/earn/components/add-edit-plan-modal/index.tsx]: Done.

**Comment by worldomonation** [comment]
Code Style build is permafailing and reporting a large number of linting issues. It seems this PR has been living for quite some time (1+ month) and the branch possibly longer than that, so I'd suggest that you pull the most recent `trunk` and rebase this branch on top. 

I wrote a p7H4VZ-3Oh-p2 on how to rebase without causing a review bomb.

image

Edit: it also seems that a large number of the changed files in this PR is just formatting change that should not have happened. I would advise that you check your ESLint configuration to make sure it isn't generating what's essentially whitespace diffs.

**Review by worldomonation — CHANGES_REQUESTED** [review]
Please see previous comment.

**Comment by olaseni** [comment]
This looks good from an `/email` perspective. I can see that the changes are really just formatting changes, and there seems to be no regression.

I did not test the other changes though.

**Comment by millerf** [comment]
@worldomonation @olaseni It looks like the merge I intended didn't go as expected. 
I'll rebase on trunk from my latest good commit now and force-push.

Thanks for the pings

**Comment by millerf** [comment]
The different pings were automatically added. I have no idea why. Sorry about the wide pings...

**Comment by millerf** [comment]
Closing in favor of https://github.com/Automattic/wp-calypso/pull/82728

**Pull request closed without merging**